internal/pkg/manifest/svc.go

```diff
@@ -570,3 +575,79 @@ func sortExposedPorts(exposedPorts []ExposedPort) []ExposedPort {
 	})
 	return exposedPorts
 }
+
+// HTTPLoadBalancerTarget returns target container and target port for the ALB configuration.
+func (s *LoadBalancedWebService) HTTPLoadBalancerTarget(exposedPorts []ExposedPort, rrTargetContainer *string, rrTargetPort *uint16) (targetContainer *string, targetPort *string) {
```

Comment:
Just some food for thought 💭 Not a request.

It seems like we are passing information from `manifest` to `stack` and then from `stack` back to `manifest`. The `exposedPort`, `rrTargetContainer` are examples.  In general, I found this pattern a bit difficult to follow.

From what I understand, for us to know:
1. The summary of port mappings (that is, `[]ExposedPort` of all containers)
2. For each container, what are the ports that it exposes (`[]ExposedPort` of one container)
3.  What is the target container, and target port

All of these information can be computed within the `manifest` package. 

Ideally, we would have a `manifest` package whose interface looks something like:
```go
type LoadBalancedWebService struct {
    cachedExposedPorts []ExposedPort
}

type ParsedSidecarConfig struct {
    Name string
    ...
    Ports []ExposedPort // Contains the ports that are exposed by the sidecar
}
func (s *LoadBalancedWebService) Sidecars() []ParsedSidecarConfig // Instead of figuring out what ports a sidecar container expose in `stack/transformer.go` L122, we do it in `manifest` pkg

func (s *LoadBalancedWebService) HTTPLoadBalancerTarget() (targetContainer, targetPort string)

```

Do you think this makes sense?

Reply:
> Why do we prefer passing rrTargetContainer and rrTargetPort as parameters?

I have removed this from the code for now which is required in the next PRs when additional rules comes into the picture. I will add that part when I will release the PR for additional rules.